File: 1903-largest-odd-number-in-string/1903-largest-odd-number-in-string.cpp
class Solution {
public:
    string largestOddNumber(string num) {
        if((int)(num[num.size()-1])%2!=0) return num;
        int n=num.size();
        int e=0;
        for(;e<n;e++){
            if((int)(num[e])%2!=0) break;
        }
        if(e==n) return "";
        e=n-2;
        while(e>=0 && (int)(num[e])%2==0) e--;
        return num.substr(0,e+1);
    }
};

File: 1903-largest-odd-number-in-string/1903-largest-odd-number-in-string.java
class Solution {
    public String largestOddNumber(String num) {
        int n=num.length();
        int e=n-1;
        while(e>=0 && (int)(num.charAt(e))%2==0) e--;
        if(e==-1) return "";
        return num.substring(0,e+1);
    }
}